lib/configuration.js

```diff
@@ -15,150 +15,429 @@
  */
 
 'use strict';
+var env = process.env;
+var inherits = require('util').inherits;
+var EventEmitter = require('events').EventEmitter;
+var commonDiag = require('@google/cloud-diagnostics-common');
+var utils = commonDiag.utils;
 var lodash = require('lodash');
 var isPlainObject = lodash.isPlainObject;
-var isString = lodash.isString;
-var isNumber = lodash.isNumber;
+var isObject = lodash.isObject;
 var isUndefined = lodash.isUndefined;
+var isFunction = lodash.isFunction;
+var isBoolean = lodash.isBoolean;
+var isString = lodash.isString;
+var isEmpty = lodash.isEmpty;
+var isNull = lodash.isNull;
 var has = lodash.has;
-var env = process.env;
 var version = require('../package.json').version;
 
 /**
- * @typedef ServiceContextConfiguration
- * @type Object
- * @property {String} service - the hosting applications underlying service
- *  platform
- * @property {String} version - the hosting applications version
+ * The Configuration constructor function initializes several internal
+ * properties on the Configuration instance and accepts a runtime-given
+ * configuration object which may be used by the Configuration instance
+ * depending on the initialization transaction that occurs with the meta-data
+ * service.
+ * @class Configuration
+ * @extends EventEmitter
```

Comment:
This no longer needs needs to be an EventEmitter, right? The following classdesc needs to be updated as well.